File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string>ans;
        int curr=0;
       for(int i=0;i<target.size();i++){
           while(curr<target[i]-1){
               ans.push_back("Push");
               ans.push_back("Pop");
               curr++;
           }
               ans.push_back("Push");
           curr++;
           
       }
        return ans;
        
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
         List<String> ans=new ArrayList<>();
        int size=target.length;
        int curr=0;
        for (int i = 0; i < size; i++) {
            while (curr<target[i]-1){
                ans.add("Push");
                ans.add("Pop");
                curr++;
            }
            ans.add("Push");
            curr++;
        }
        return ans;
    }
}